localstack/aws/handlers/auth.py

```diff
@@ -33,18 +37,21 @@ class AccountIdEnricher(Handler):
     """
 
     def __call__(self, chain: HandlerChain, context: RequestContext, response: Response):
-        access_key_id = extract_access_key_id_from_auth_header(context.request.headers)
-
-        if not access_key_id:
-            access_key_id = TEST_AWS_ACCESS_KEY_ID
-
-        # Save the request access key ID in the current thread local storage
+        # Obtain the access key ID and save it in the thread context
+        access_key_id = (
+            extract_access_key_id_from_auth_header(context.request.headers)
+            or TEST_AWS_ACCESS_KEY_ID
```

Comment:
`or TEST_AWS_ACCESS_KEY_ID`

When is the fallback here necessary? Are there any reasonable cases that need this?

Reply:
IIRC there are certain S3 calls where `extract_access_key_id_from_auth_header()` does not apply. I don't remember off the top of my head but can look up if needed